### PR TITLE
Support for azurerm_container_group data source

### LINKED ISSUE #14945: Support for azurerm_container_group data source
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In the current version a data source for Azure Container Group Instances is missing. There's only the ressource type.

### New or Affected Resource(s)

* azurerm_container_group

### Potential Terraform Configuration

```hcl
provider "azurerm" {
  features {}
  subscription_id = var.subscription_id
}

data "azurerm_container_group" "test" {
  name = "test-container-group"
  resource_group_name = var.rg.name
}

output "ip_address" {
  value = data.azurerm_container_group.test.ip_address
}

output "fqdn" {
  value = data.azurerm_container_group.test.fqdn
}
```

### References

* PR: #14946

### PR DESCRIPTION
**Pull request by logineimer**
Fixes #14945. Let me know if there's something missing.

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
Have a test failure @logineimer 
```
------- Stdout: -------
=== RUN   TestAccDataSourceAzureRMContainerGroup_basic
=== PAUSE TestAccDataSourceAzureRMContainerGroup_basic
=== CONT  TestAccDataSourceAzureRMContainerGroup_basic
    testcase.go:110: Step 1/1 error: Check failed: Check 5/5 error: data.azurerm_container_group.test: Attribute 'fqdn' expected to be set
--- FAIL: TestAccDataSourceAzureRMContainerGroup_basic (98.88s)
FAIL
```

**Comment by logineimer** [comment]
Yeah my bad, the dns_label_name was missing in the test ressources. This led to the fqdn being an empty string. Now it should work.

**Comment by katbyte** [comment]
Tests pass now, but linting needs fixing - `make terrafmt` should fix that up

**Comment by logineimer** [comment]
I fixed the linting error. Looking forward to merge. :)

File: internal/services/containers/container_group_data_source.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+func dataSourceContainerGroupRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Containers.GroupsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	resp, err := client.Get(ctx, resourceGroup, name)
```
Comment (stephybun) [inline internal/services/containers/container_group_data_source.go:58]: We should generate the container group ID like we do in the resource
```suggestion
	id := parse.NewContainerGroupID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))

	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
```

File: internal/services/containers/container_group_data_source.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Container Group %q was not found in Resource Group %q", name, resourceGroup)
+		}
+
+		return fmt.Errorf("making Read request on Azure Container Group %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
```
Comment (stephybun) [inline internal/services/containers/container_group_data_source.go:69]: And here we can use the values from the id
```suggestion
	d.SetId(id.ID())
	d.Set("name", id.Name)
	d.Set("resource_group_name", id.ResourceGroup)
```

File: website/docs/d/container_group.html.markdown [hunk 3]
```diff
@@ -0,0 +1,60 @@
+---
+subcategory: "Container"
+layout: "azurerm"
+page_title: "Azure Resource Manager: Data Source: azurerm_container_group"
```
Comment (stephybun) [inline website/docs/d/container_group.html.markdown:4]:
```suggestion
page_title: "Azure Resource Manager: azurerm_container_group"
```

**Comment by logineimer** [comment]
Thanks for the feedback! I never used Go before and oriented my solution at similar resources and data source. Would you like me to fixup the new commits now that I applied your suggestions? :)

File: internal/services/containers/container_group_data_source.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
```
Comment (stephybun) [inline internal/services/containers/container_group_data_source.go:13]:
```suggestion
	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
	"github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/parse"
	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
	"github.com/hashicorp/terraform-provider-azurerm/utils"
```

File: internal/services/containers/container_group_data_source.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+			},
+
+			"tags": tags.SchemaDataSource(),
+		},
+	}
+}
+
+func dataSourceContainerGroupRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Containers.GroupsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
```
Comment (stephybun) [inline internal/services/containers/container_group_data_source.go:52]:
```suggestion
	client := meta.(*clients.Client).Containers.GroupsClient
	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for making those changes @logineimer! Looks like we forgot to import the parse package and initialize the `subscriptionId` needed for the parser, one those are there I can run the tests.

**Comment by logineimer** [comment]
No problem @stephybun! Just to be sure: Do you want the commits to be squashed? :)

**Review by stephybun — APPROVED** [review]
Thanks @logineimer, LGTM!
Don't worry about squashing, I can do that when I merge 🚀

**Pull request merged**